drivers/ethernet/eth_w5500.c

```diff
@@ -502,14 +502,7 @@ static void w5500_set_macaddr(const struct device *dev)
 	struct w5500_runtime *ctx = dev->data;
 
 #if DT_INST_PROP(0, zephyr_random_mac_address)
-	/* override vendor bytes */
-	memset(ctx->mac_addr, '\0', sizeof(ctx->mac_addr));
-	ctx->mac_addr[0] = WIZNET_OUI_B0;
-	ctx->mac_addr[1] = WIZNET_OUI_B1;
-	ctx->mac_addr[2] = WIZNET_OUI_B2;
-	if (ctx->generate_mac) {
-		ctx->generate_mac(ctx->mac_addr);
-	}
+	gen_random_mac(ctx->mac_addr, WIZNET_OUI_B0, WIZNET_OUI_B1, WIZNET_OUI_B2);
```

Comment:
can you convert this code to work for all instances?

Reply:
W5500 is not currently a multi-instance driver, it uses `ETH_NET_DEVICE_DT_INST_DEFINE(0, ...)` rather than `DT_INST_FOREACH_STATUS_OKAY`.

It would not be difficult to make it multi-instance, should I do that as part of this PR?

Reply:
My mistake I'm not sure what I got mixed up there